### DISCUSSION TITLE
How can I create a catalogue from verdaccio 5?

### DISCUSSION THREAD

**Discussion by AllanOricil** [verdaccio/verdaccio#4054] [Q&A]
I want to create a JSON with all packages that are currently stored in my verdaccio storage. I was able to do it in Verdaccio 4 because there is a route `/-/all` but in 5 this route was removed. So how can I fetch a list of all package that are locale stored and managed by Verdaccio 5?

**Comment by juanpicado** [comment]
The endpoint all was meant for search but was removed time ago since npmjs also deprecated, thus I don't expect anything similar to have again.

**Reply by AllanOricil** [reply]
Do you know a way I can retrieve all packages that are stored in verdaccio in a json format?

**Reply by AllanOricil** [reply]
Do you know a way I can retrieve all packages that are stored in verdaccio in a json format?

**Reply by AllanOricil** [reply]
@juanpicado is there a plugin ? If not, could you help me with the initial steps to implement one that can read the database, create the catalogue in a given format, and expose it in a given route?

**Reply by AllanOricil** [reply]
I'm thinking I'm going to have to create a middleware plugin. How can I get the list of packages stored in verdaccio using the `storage` object? Could not find it.



https://verdaccio.org/docs/plugin-middleware

**Reply by AllanOricil** [reply]
I think I know how to do it. I need to use 


`storage.getLocalDatabase()` right?

**Reply by jchown** [reply]
@AllanOricil did you manage to do this?

**Reply by AllanOricil** [reply]
@jchown I did not have time yet. If you know how to try my suggestion, could you do it and share your results here? Thanks
